state/spacesdiscovery.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/utils/set"
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/juju/environs"
+	"github.com/juju/juju/instance"
+	"github.com/juju/juju/network"
+)
+
+func (st *State) getModelSubnets() (set.Strings, error) {
+	subnets, err := st.AllSubnets()
```

Comment:
I didn't go through any of this code thoroughly as I understand it is just ported from the old worker, right?
I believe we talked about a couple things that weren't great with the old code, so if you want me to look it over, I can, but please ask.